tests/test_restrictions.py

```diff
@@ -0,0 +1,143 @@
+import pandas as pd
+from pandas.util.testing import assert_series_equal
+
+from zipline.finance.restrictions import (
+    RESTRICTION_STATES,
+    Restriction,
+    InMemoryRestrictions,
+    StaticRestrictions,
+    NoopRestrictions,
+)
+from zipline.testing.fixtures import (
+    WithDataPortal,
+    ZiplineTestCase,
+)
+
+str_to_ts = lambda dt_str: pd.Timestamp(dt_str, tz='UTC')
+
+
+class RestrictionsTestCase(WithDataPortal, ZiplineTestCase):
+
+    ASSET_FINDER_EQUITY_SIDS = 1, 2, 3
+
+    @classmethod
+    def init_class_fixtures(cls):
+        super(RestrictionsTestCase, cls).init_class_fixtures()
+        cls.ASSET1 = cls.asset_finder.retrieve_asset(1)
+        cls.ASSET2 = cls.asset_finder.retrieve_asset(2)
+        cls.ASSET3 = cls.asset_finder.retrieve_asset(3)
+
+    def test_in_memory_restrictions(self):
+
+        restrictions = [
+            Restriction(
+                self.ASSET1,
+                str_to_ts('2011-01-06'),
+                RESTRICTION_STATES.FROZEN
+            ),
+            Restriction(
+                self.ASSET1,
+                str_to_ts('2011-01-04'),
+                RESTRICTION_STATES.FROZEN
+            ),
+            Restriction(
+                self.ASSET1,
+                str_to_ts('2011-01-05'),
+                RESTRICTION_STATES.ALLOWED
+            ),
+            Restriction(
+                self.ASSET2,
+                str_to_ts('2011-01-03'),
+                RESTRICTION_STATES.FROZEN
+            ),
+        ]
+
+        rlm = InMemoryRestrictions(restrictions)
+
+        # Check the restrictions are ordered by dt for each sid
+        self.assertEqual(
+            [r.effective_date for r in
+             rlm._restrictions_by_asset[self.ASSET1]],
+            [str_to_ts(dt_str) for dt_str in
+             ('2011-01-04', '2011-01-05', '2011-01-06')]
+        )
+
+        self.assertFalse(
```

Comment:
As a reader, it's hard for me to see the relationship between the assertions being made here and the original test data.  I would write this as something like:

```python
    def test_historical_restrictions(self):
        A1_FREEZE = str_to_ts('2011-01-04')
        A1_THAW = str_to_ts('2011-01-05')
        A1_RE_FREEZE = str_to_ts('2011-01-06')
        minute = pd.Timedelta('1 minute')

        # Build restrictions.

        def assert_is_restricted(asset, dt):
            self.assertTrue(rlm.is_restricted(asset, dt))

        def assert_not_restricted(asset, dt):
            self.assertFalse(rlm.is_restricted(asset, dt))

        # Not restricted until on or after the freeze.
        assert_not_restricted(self.ASSET1, A1_FREEZE - one_minute)
        assert_is_restricted(self.ASSET1, A1_FREEZE)
        assert_is_restricted(self.ASSET1, A1_FREEZE + one_minute)

        # Unrestricted on or after the thaw.
        assert_is_restricted(self.ASSET1, A1_THAW - one_minute)
        assert_not_restricted(self.ASSET1, A1_THAW)
        assert_not_restricted(self.ASSET1, A1_THAW + one_minute)

        # Restricted again on or after the freeze.
        assert_not_restricted(self.ASSET1, A1_RE_FREEZE - one_minute)
        assert_is_restricted(self.ASSET1, A1_RE_FREEZE)
        assert_is_restricted(self.ASSET1, A1_RE_FREEZE + one_minute)
        # Should stay restricted for the rest of time.
        assert_is_restricted(self.ASSET1, A1_RE_FREEZE + one_minute * 100000)

         # Assertions for ASSET2.
```

The above also has the benefit that we could parametrize on the freeze/thaw dates without affecting the business logic of the test.